auth-server/src/main/java/com/epam/petclinic/authentication/AuthorizationServerConfig.java

```diff
@@ -0,0 +1,66 @@
+package com.epam.petclinic.authentication;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.oauth2.config.annotation.configurers.ClientDetailsServiceConfigurer;
+import org.springframework.security.oauth2.config.annotation.web.configuration.AuthorizationServerConfigurerAdapter;
+import org.springframework.security.oauth2.config.annotation.web.configuration.EnableAuthorizationServer;
+import org.springframework.security.oauth2.config.annotation.web.configurers.AuthorizationServerEndpointsConfigurer;
+import org.springframework.security.oauth2.provider.token.TokenStore;
+import org.springframework.security.oauth2.provider.token.store.JwtAccessTokenConverter;
+import org.springframework.security.oauth2.provider.token.store.JwtTokenStore;
+import org.springframework.security.oauth2.provider.token.store.KeyStoreKeyFactory;
+
+@Configuration
+@EnableAuthorizationServer
+public class AuthorizationServerConfig extends AuthorizationServerConfigurerAdapter {
+
+    @Autowired
+    @Qualifier("authenticationManagerBean")
+    private AuthenticationManager authenticationManager;
+
+    @Override
+    public void configure(ClientDetailsServiceConfigurer clients) throws Exception {
+        clients.inMemory()
+                .withClient("ui")
+                .secret("123")
+                .resourceIds("ORDER-SERVICE", "CLINIC-SERVICE", "GATEWAY-SERVICE")
+                .scopes("create", "read", "update", "delete")
+                .autoApprove(true)
+                .authorities("admin")
+                .authorizedGrantTypes("implicit", "refresh_token", "password", "authorization_code",
+                        "client_credentials")
+                .refreshTokenValiditySeconds(600)
+                .accessTokenValiditySeconds(300);
+    }
+
+    /**
+     * Configures the OAuth2 endpoints to adjust the authentication manager which will represent the web-security users,
+     * and JWT token store configuration.
+     *
+     * @param endpoints @see AuthorizationServerEndpointsConfigurer.
+     * @throws Exception @see Exception on any error.
+     */
+    @Override
+    public void configure(AuthorizationServerEndpointsConfigurer endpoints) throws Exception {
+        endpoints.tokenStore(tokenStore()).tokenEnhancer(jwtTokenEnhancer()).authenticationManager(authenticationManager);
```

Comment:
Different instances of tokenEnhancer for tokenStore and tokenEnhencer of endpoint?